kratos/utilities/function_parser_utility.cpp

```diff
@@ -12,6 +12,9 @@
 //
 
 // System includes
+#ifdef KRATOS_MODERN_REGEX
+    #include <regex>
+#endif
```

Comment:
```suggestion
#if defined __has_include
    if __has_include(<regex>)
        #define KRATOS_MODERN_REGEX
    #endif
#endif

#ifdef KRATOS_MODERN_REGEX
    #include <regex>
#endif
```

this way would be more modern and automatic
or sth in that direction

Reply:
`__has_include` is officially C++17, but already supported by most compilers afaik
https://en.cppreference.com/w/cpp/preprocessor/include

Reply:
Is not a problem of compilation, is a problem of full support (apparently in 4.8, the regex support is partial)

Reply:
ah ok
hm maybe then the following:
```suggestion
#if !(defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 4 || (__GNUC__ == 4 && (__GNUC_MINOR__ < 9))))
    #include <regex>
#endif
```

from here:
https://github.com/KratosMultiphysics/Kratos/blob/9ea10775243e921f8caaa115eaca4a235ab9cb4d/applications/HDF5Application/custom_io/hdf5_file.cpp#L22

Reply:
So, you want to avoid the KRATOS_MODERN_REGEX macro?

Reply:
yes

Reply:
Do I also remove it?